relayer/src/chain.rs

```diff
@@ -67,7 +67,7 @@ pub struct QueryPacketOptions {
 }
 
 /// Defines a blockchain as understood by the relayer
-pub trait Chain: Sized {
+pub trait ChainEndpoint: Sized {
```

Comment:
I have renamed this trait to `ChainEndpoint` so that we can unambiguously use the type name `Chain` as type variable. Let me know if you have a better name for this trait.

Reply:
The `Chain` trait is a representation of a IBC-enabled chain from the perspective of the relayer. I think `ChainEndpoint` makes sense, but I can think of one drawback: it gives the impression of a single point of contact (and that's not accurate, as the relayer interfaces with each chain via potentially multiple points, e.g., full nodes, witnesses).

The alternative `IbcChain` or `IbcEnabledChain` would be most accurate in my opinion, but that seems clunky. We can keep `ChainEndpoint` as far as I'm concerned.